### ISSUE TITLE
All operators visible despite filter

### ISSUE THREAD

**Issue by dadav** [openshift/oc-mirror#930]
# Version

```console
$ oc-mirror version
Client Version: version.Info{Major:"", Minor:"", GitVersion:"4.16.0-202407221538.p0.g2bfefcc.assembly.stream.el9-2bfefcc", GitCommit:"2bfefccc11eb7f08ce6b1806624e03472bf33b6e", GitTreeState:"clean", BuildDate:"2024-07-22T16:12:18Z", GoVersion:"go1.21.11 (Red Hat 1.21.11-1.module+el8.10.0+21986+2112108a) X:strictfipsruntime", Compiler:"gc", Platform:"linux/amd64"}
```

# What happened?

I used the following config to make only the neuvector available in the cluster:

```yaml
operators:
    - catalog: registry.redhat.io/redhat/certified-operator-index:v4.16
      full: true
      packages:
        - name: neuvector-certified-operator
```

Which was successful, but now every certified operator is visible in the operatorhub. You can't install them, because the images weren't mirrored. This confuses the users, because they don't know which operators really can be installed.

# What did you expect to happen?

The users can only see the neuvector operator in the operatorhub.

**Comment by dadav** [comment]
In the past, only the operators listed in `packages` were visible in the operatorhub. I don't know if this is intentional, but I'd love to have this back.

**Comment by dadav** [comment]
It's basically this example:

https://github.com/openshift/oc-mirror/blob/main/docs/examples/imageset-config-filter-catalog.yaml#L29

**Comment by sherine-k** [comment]
Hi @dadav 
This is now fixed in 4.17 of OCP
